packages/fetch-and-compile/lib/types.ts

```diff
@@ -0,0 +1,38 @@
+import type { WorkflowCompileResult } from "@truffle/compile-common";
+import type { SourceInfo } from "@truffle/source-fetcher";
+
+export type FailureType = "fetch" | "compile";
+
+export interface SingleResult {
```

Comment:
```suggestion
export interface Result {
```

Or we could call it `FetchAndCompileResult`?

Reply:
Well, it's called `SingleResult` because it's for a single address.  Remember I was originally going to write this to handle multiple addresses.  I don't want to just call it `Result`, or `FetchAndCompileResult`, because in the future we might add other forms of result.  Maybe the name should be `SingleAddressResult` to be more explicit there?

Reply:
in my mind, the SingleResult is the normal case; the debugger way is the weird one. so my thinking is: name the normal type as something normal-sounding; don't confuse people into thinking there's a special case here when none exists.

I see your point though, but still... `SingleAddressResult` is better, but why distinguish prematurely? We can always shim it later if we add a new kind of result.

Reply:
I mean I wasn't contrasting it against the existing debugger case, but against a hypothetical multi-address case.

Reply:
I don't think it's worth making that distinction now... since this is externally facing, it strikes me as noise because the user may distract themselves looking for a multi-result that doesn't exist yet.

Reply:
OK, cool, will do.

Reply:
anyway, later we can make `FetchAndCompileResult` be polymorphic like we do in all the other packages.

or if you're worried about it, an alternative is to plan the multi-address result form now and just have single-result conform to that from the get-go. no distinction needed.

Reply:
Nah, keeping it simple, it'll just be `FetchAndCompileResult`.